### PR TITLE
Waypoint docs landing page updates

### PR DESCRIPTION
**Pull request by tunzor**
## ✅ "Ready for Review" checklist

- [x] The Vercel preview link has been added to this PR's description
- [x] The Asana task has been added to this PR's description
- [x] The description template has been filled in below

---

## 🔗 Relevant links

- [Preview link](https://dev-portal-git-arwaypoint-docs-landing-hashicorp.vercel.app/waypoint/docs) 🔎
- [Asana task](https://app.asana.com/0/1201979547919485/1201979547919596/f) 🎟️

## 🗒️ What

Updates to the Waypoint docs page on devdot highlighting specific references pages.

## 🤷 Why

Docs page should have some commonly referenced pages and features highlighted.

## 📸 Design Screenshots

image

### CODE REVIEW

**Review by BrandonRomano — APPROVED** [review]
Looks good from my end: I think probably should get @topfunky's 🟢 too before merging!

**Comment by ashleemboyer** [comment]
@BrandonRomano some quick context because @tunzor have been chatting via DM, I'm going to make a couple changes he requested on this branch in a little bit!

**Comment by ashleemboyer** [comment]
@tunzor I just pushed up the code changes that should make the MDX content on Waypoint's /docs page show after the marketing content! I also added some documentation the `mdxSlot` so it's front-and-center for when this view is added for future products. Let me know if there are any clarifications I can make there!
- New docs added in this commit: https://github.com/hashicorp/dev-portal/pull/477/commits/9b66dbfeb843ec9dbc5a25e372d140ee09d5826d.

@BrandonRomano I'm re-requesting your review to get engineering eyes on the small changes I made to re-order the content sections and update the sidecar to reflect that as well.
- Changes made in this commit: https://github.com/hashicorp/dev-portal/pull/477/commits/66858f99009816defdceaded496f68d9ea1677a4

---

To test my changes:

- [ ] Go to https://dev-portal-git-arwaypoint-docs-landing-hashicorp.vercel.app/waypoint/docs
- [ ] Make sure the marketing content comes before the MDX content
- [ ] Make sure all of the sidecar links jump to the correct heading in the page

**Review by topfunky — APPROVED** [review]
Looks great for the content!

Did you mean to include edits to `server.ts`, `index.ts`, etc.?

File: src/views/product-root-docs-path-landing/product-root-docs-path-landing.module.css [hunk 1]
```diff
@@ -1,9 +1,5 @@
 .mdxSlotWrapper {
-  margin-bottom: 48px;
-
-  &:last-child {
-    margin-bottom: 0;
-  }
+  margin-top: 48px;
```
Comment (ashleemboyer) [inline src/views/product-root-docs-path-landing/product-root-docs-path-landing.module.css:2]: 🗒️ Quick note: these changes in this file were made by me

**Review by ashleemboyer — COMMENTED** [review]
Should have done this to start to make it clearer which changes were mine, but I've added some annotations to hopefully prevent any confusion about which changes were made by who! 🙏🏻

**Pull request merged**